indico/modules/events/editing/controllers/backend/timeline.py

```diff
@@ -50,17 +50,23 @@ def get_file_context(self):
         return 'event', self.event.id, 'editing', self.contrib.id, self.editable_type.name
 
 
-class RHEditingUploadPaperLastRevision(RHEditingUploadFile):
+class RHEditingUploadContributionFile(RHEditingUploadFile):
     @use_kwargs({
-        'id': fields.Int()
+        'id': fields.Int(load_default=None),
+        'paper_id': fields.Int(load_default=None)
     })
-    def _process(self, id):
-        last_rev = self.contrib.paper.get_last_revision()
-        if last_rev:
-            found = next((f for f in last_rev.files if f.id == id), None)
-            if found:
-                with found.open() as stream:
-                    return self._save_file(found, stream)
+    def _process(self, id, paper_id):
+        if bool(id) == bool(paper_id):
+            raise BadRequest
+        files = []
+        if id and self.contrib.editables:
+            files = [file.file for e in self.contrib.editables
+                     if e.type == self.editable.type for file in e.revisions[-1].files]
+        if paper_id and self.contrib.paper and (last_rev := self.contrib.paper.get_last_revision()):
+            files.extend(last_rev.files)
```

Comment:
Why not `files = ...` since we never expect to get files from multiple sources?

Reply:
Initially, I was expecting to have both but didn't update accordingly.